src/System.Memory/src/System/Buffers/IMemoryList.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a linked list of <see cref="Memory{T}"/> nodes.
+    /// </summary>
+    public interface IMemoryList<T>
+    {
+        /// <summary>
+        /// The <see cref="Memory{T}"/> value for current nodes.
+        /// </summary>
+        Memory<T> Memory { get; }
+
+        /// <summary>
+        /// The next node.
+        /// </summary>
+        IMemoryList<T> Next { get; }
+
+        /// <summary>
+        /// The sum of node lenghts before current.
```

Comment:
typo -> lenghts

Can you explain further in the comments what you mean by `before current`?